sql/cache_test.go

```diff
@@ -0,0 +1,166 @@
+package sql
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCacheKey(t *testing.T) {
+	k := CacheKey(1)
+	require.Equal(t, uint64(0x4320000000000000), k)
+}
+
+func TestLRUCache(t *testing.T) {
+	t.Run("basic methods", func(t *testing.T) {
+		require := require.New(t)
+
+		cache := newLRUCache(mockMemory{}, fixedReporter(5, 50), 10)
+
+		require.NoError(cache.Put(1, "foo"))
+		v, err := cache.Get(1)
+		require.NoError(err)
+		require.Equal("foo", v)
+
+		_, err = cache.Get(2)
+		require.Error(err)
+		require.True(ErrKeyNotFound.Is(err))
+
+		// Free the cache and check previous entry disappeared.
+		cache.Free()
+
+		_, err = cache.Get(1)
+		require.Error(err)
+		require.True(ErrKeyNotFound.Is(err))
+
+		cache.Dispose()
+		require.Panics(func() {
+			cache.Get(1)
+		})
+	})
+
+	t.Run("no memory available", func(t *testing.T) {
+		require := require.New(t)
+		cache := newLRUCache(mockMemory{}, fixedReporter(51, 50), 5)
+
+		require.NoError(cache.Put(1, "foo"))
+		_, err := cache.Get(1)
+		require.Error(err)
+		require.True(ErrKeyNotFound.Is(err))
+	})
+
+	t.Run("free required to add entry", func(t *testing.T) {
+		require := require.New(t)
+		var freed bool
```

Comment:
Maybe we can explicitly assert that this variable is set to true after the Put? I understand that is implicit in the Get giving no error, but anyway. Same for the other tests below.

Reply:
Good catch. It was the intention of that variable but forgot to assert it later 😅